### PR TITLE
Remove XPRESS Interface

### PR DESCRIPTION
**Pull request by mrmundt**
## Summary/Motivation:
Based on a discussion in this week's developer's call, this removes the `XPRESS` solver interface in favor of `xpress_direct`.

## Changes proposed in this PR:
- Remove `XPRESS.py`
- Replace `xpress` with `xpress_direct` in testing

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by michaelbynum — APPROVED** [review]
Looks good (pending tests).

**Review by blnicho — APPROVED** [review]

**Comment by blnicho** [comment]
With this PR I think we can close several open issues related to this XPRESS interface. There is a list of associated issues in PR #2125. Are there any objections to closing the following issues: #58, #59, #69, #1429, #1601

I assume these issues don't appear in the persistent and direct interfaces. @bknueven I think you were the one to mark the old XPRESS interface as broken. Any input on this?

**Comment by bknueven** [comment]
On my system this morning, trying to use the Xpress LP interface:
```
Python 3.7.7 (default, Mar 26 2020, 10:32:53) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.13.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from pyomo.environ import *; m = ConcreteModel(); m.x = Var(); m.c = Constraint(expr=(0., m.x, 1.)); m.o = Objective(expr=m.x)                    

In [2]: s = SolverFactory('xpress')                                                                                                                       
WARNING: The shell interface for Xpress is broken for recent versions of
    Xpress. Please use xpress_direct or xpress_persistent, which require the
    Xpress Python API. Python bindings for recent versions of Xpress can be
    installed via `pip`: <https://pypi.org/project/xpress/>.

In [3]: s.available()                                                                                                                                     
Out[3]: True

In [4]: r = s.solve(m, tee=True)                                                                                                                          
FICO Xpress Solver 64bit v8.8.0 Nov 21 2019
(c) Copyright Fair Isaac Corporation 1983-2019. All rights reserved
 Optimizer v35.01.01    [/Applications/FICO Xpress/Xpress Workbench.app/Contents/Resources/xpressmp/lib/libxprs.dylib]
 
Reading Problem /tmp3x8zp7mk.pyomo
Problem Statistics
           3 (      0 spare) rows
           2 (      0 spare) structural columns
           3 (      0 spare) non-zero elements
Global Statistics
           0 entities        0 sets        0 set members
FICO Xpress v8.8.0, Hyper, solve started 8:14:43, Jan 31, 2022
Heap usage: 83KB (peak 83KB, 582KB system)
Minimizing LP /tmp3x8zp7mk.pyomo
Original problem has:
         3 rows            2 cols            3 elements
Presolved problem has:
         0 rows            0 cols            0 elements
Presolve finished in 0 seconds
Heap usage: 84KB (peak 102KB, 583KB system)
 
   Its         Obj Value      S   Ninf  Nneg   Sum Dual Inf  Time
     0           .000000      D      0     0        .000000     0
Uncrunching matrix
Optimal solution found
Dual solved problem
  0 simplex iterations in 0s

Final objective                       : 0.000000000000000e-01
  Max primal violation      (abs/rel) :       0.0 /       0.0
  Max dual violation        (abs/rel) :       0.0 /       0.0
  Max complementarity viol. (abs/rel) :       0.0 /       0.0
All values within tolerances

In [5]: assert_optimal_termination(r)                                                                                                                     
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-5-cdb467db575a> in <module>
----> 1 assert_optimal_termination(r)

~/Software/pyomo/pyomo/opt/results/solver.py in assert_optimal_termination(results)
    162             'Solver status: {}, Termination condition: {}'.format(results.solver.status,
    163                                                                   results.solver.termination_condition)
--> 164         raise RuntimeError(msg)
    165 
    166 

RuntimeError: Solver failed to return an optimal solution. Solver status: ok, Termination condition: unknown

In [6]: import pyomo                                                                                                                                      

In [7]: pyomo.__version__                                                                                                                                 
Out[7]: '6.2.1.dev0'                                                                                                                  
```

While I don't have the issues with this simple model reported elsewhere (though I have seen those pop up too), even a basic thing like returning correct termination codes is broken. Therefore, I think removing this interface is a good change, unless somebody has the time and resources to put into fixing the existing LP interface. (I don't at the moment.)

However, with this PR as it is, `SolverFactory("xpress")` will return an `UnknownSolver` object, which is perhaps less helpful to Xpress users than the current behavior. If this shouldn't point to `XpressDirect`, (perhaps in case the LP interface gets revived some day), could we keep a stub of the current interface to point Xpress users in the right direction?

File: pyomo/solvers/plugins/solvers/XPRESS.py [hunk 1]
```diff
@@ -35,421 +15,19 @@ def __new__(cls, *args, **kwds):
         try:
             mode = kwds['solver_io']
             if mode is None:
-                mode = 'lp'
+                mode = 'python'
             del kwds['solver_io']
         except KeyError:
-            mode = 'lp'
+            mode = 'python'
```
Comment (jsiirola) [inline pyomo/solvers/plugins/solvers/XPRESS.py:21]: Would the following would be cleaner / clearer?
```python
mode = kwds.pop('solver_io', 'python')
if mode is None:
    mode = 'python'
```

**Review by jsiirola — APPROVED** [review]
One minor change [which I will push an edit for momentarily].

File: pyomo/solvers/plugins/solvers/XPRESS.py (hunk 1 above)
Comment (michaelbynum) [reply pyomo/solvers/plugins/solvers/XPRESS.py]: Thanks @jsiirola. Great change.

**Comment by michaelbynum** [comment]
The failing test is intermittent and unrelated. I am going to merge this.

**Pull request merged**